### PR TITLE
MCLOUD-5837: HA load balancer issues

### PR DESCRIPTION
**Pull request by dhorytskyi**
### Description
When uploading images to the image gallery from the Pagebuilder interface the store will return a "Destination folder is not writable or does not exist" error.
This issue is sporadic and does not always occur.

Cloud shares file-systems between nodes where the replication is slow: there is a noticeable lag between the creation of a file on one node and its appearance on the other nodes, which is readily tested.
There is load balancer between the nodes which arbitrary redirects queries of a session to other nodes which created problems, for instance uploading images to the image gallery from Pagebuilder.

### Fixed Issues (if relevant)
https://jira.corp.magento.com/browse/MCLOUD-5837

### Manual testing scenarios
1. Go Content>Pages
2. Select Edit for any page
3. Select Content
4. In the content section, choose Select from Gallery
5. Click Browse
6. Choose an image
7. Click Open

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

releasenotes
Fixes a Page Builder interface issue that sometimes caused the following error when uploading images to the image gallery: `Destination folder is not writable or does not exist`

### CODE REVIEW

File: patches/MCLOUD-5837__fix_filesystem_load_balancer_issue.patch [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
++        } elseif (!$this->getFileDriver()->isWritable($destinationFolder)) {
+             throw new FileSystemException(__('Destination folder is not writable or does not exists.'));
+         }
+     }
+@@ -655,7 +666,7 @@ class Uploader
+      * @return \Magento\Framework\File\Uploader
+      * @throws FileSystemException
+      */
+-    private function _createDestinationFolder($destinationFolder)
++    private function createDestinationFolder(string $destinationFolder)
```
Comment (BarnyShergold) [inline patches/MCLOUD-5837__fix_filesystem_load_balancer_issue.patch]: Missing return type after prototype declaration

**Review by BarnyShergold — CHANGES_REQUESTED** [review]

**Comment by BarnyShergold** [comment]
@dhorytskyi - As some of the Maintainers (myself included) on Cloud do not have access Magento JIRA tickets - please add the user story of the ticket to the description of this PR

**Comment by dhorytskyi** [comment]
@BarnyShergold I've updated the description.

File: patches/MCLOUD-5837__fix_filesystem_load_balancer_issue.patch (hunk 1 above)
Comment (dhorytskyi) [reply patches/MCLOUD-5837__fix_filesystem_load_balancer_issue.patch]: This patch is based on code changes that are delivering to the mainline. So it's better to keep changes as is.

Comment (BarnyShergold) [reply patches/MCLOUD-5837__fix_filesystem_load_balancer_issue.patch]: Okay that's acceptable

**Review by BarnyShergold — DISMISSED** [review]

File: patches.json [hunk 2]
```diff
@@ -241,6 +241,9 @@
     "Sitemap Generation Warnings": {
       ">=2.3.0 <2.3.2": "MCLOUD-3025__sitemap_generation_warnings__2.3.0.patch",
       ">=2.3.2 <2.3.5": "MCLOUD-3025__sitemap_generation_warnings__2.3.3.patch"
+    },
+    "Fix load balancer issue": {
+      ">=2.3.4 <2.3.6": "MCLOUD-5837__fix_filesystem_load_balancer_issue.patch"
```
Comment (NadiyaS) [inline patches.json]: To follow patch naming standard, can you please add number from what version this patch should be applied? It should follow after double underscore at he end of file name - `MCLOUD-5837__fix_filesystem_load_balancer_issue__2.3.4.patch`

Comment (dhorytskyi) [reply patches.json]: Done

**Review by NadiyaS — APPROVED** [review]

**Review by BarnyShergold — APPROVED** [review]

**Review by oshmyheliuk — APPROVED** [review]

**Comment by chittima** [comment]
QA approved

**Comment by andriyShevtsov** [comment]
This PR was tested by @chittima. I just verified that she did everything correct. QA done here. Thanks for your help.

**Comment by andriyShevtsov** [comment]
Please don’t merge this PR yet.  We should make additional testing.

**Comment by andriyShevtsov** [comment]
Looks like we have an issue 
@dhorytskyi @chittima may you please provide some details if you. have something to add?

**Comment by andriyShevtsov** [comment]
QA passed. Thanks @chittima

**Pull request merged**